api.c

```diff
@@ -572,10 +571,6 @@ int tcmu_emulate_evpd_inquiry(
 		/* Optimal xfer length */
 		memcpy(&data[12], &val32, 4);
 
-		/* MAXIMUM WRITE SAME LENGTH */
-		val64 = htobe64(VPD_MAX_WRITE_SAME_LENGTH);
-		memcpy(&data[36], &val64, 8);
-
```

Comment:
Oh yeah, for the first part of your comment I had seen the code and commit comments you are referring to. In the git commit it had said:

> We only enable transfers bigger than 0xFFFF if the device explicitly reports MAXIMUM WRITE SAME LENGTH in the BLOCK LIMITS VPD.

And so if the VPD length is 0x3c then the device is reporting a MAX WS LENGTH in my opinion, but maybe I am wrong. If it did not want to report a WS MAX LENGTH setting, then it would set the VPD data length to a shorter value. However, I was not sure if that is what they meant by "explicitly" reporting a value.

My question before was, if it does count as explicitly reporting a valid, did they mean to handle zero as a valid  reported value or did they just miss the part in the specs about zero == unlimited, or does zero not count as a valid reported value.